ext/phar/phar_object.c

```diff
@@ -2217,6 +2217,15 @@ static zend_object *phar_rename_archive(phar_archive_data **sphar, char *ext) /*
 	zend_call_known_instance_method_with_1_params(ce->constructor, Z_OBJ(ret), NULL, &arg1);
 	zval_ptr_dtor(&arg1);
 	return Z_OBJ(ret);
+
+err_reused_oldpath:
+	if (pphar == phar) {
+		/* NOTE: we know it's not the last reference because the phar is reused. */
+		phar->refcount--;
+	}
+err_oldpath:
```

Comment:
Maybe add a comment that the fallthrough is intentional?

Reply:
I usually don't do this when I write goto-style error-handling code, but perhaps I should. I'll add it here.